core/src/main/java/com/graphhopper/routing/util/EncodingManager.java

```diff
@@ -238,13 +235,70 @@ private FlagEncoder getEncoder(String name, boolean throwExc) {
     /**
      * Determine whether a way is routable for one of the added encoders.
      */
-    public long acceptWay(ReaderWay way) {
-        long includeWay = 0;
+    public boolean acceptWay(ReaderWay way, AcceptWay acceptWay) {
+        if (!acceptWay.isEmpty())
+            throw new IllegalArgumentException("AcceptWay must be empty");
+
         for (AbstractFlagEncoder encoder : edgeEncoders) {
-            includeWay |= encoder.acceptWay(way);
+            acceptWay.put(encoder.toString(), encoder.getAccess(way));
         }
+        return acceptWay.hasAccepted();
+    }
 
-        return includeWay;
+    public static class AcceptWay {
+        private Map<String, Access> accessMap;
+        boolean hasAccepted = false;
+
+        public AcceptWay() {
+            this.accessMap = new HashMap<>(5);
+        }
+
+        private Access get(String key) {
+            Access res = accessMap.get(key);
+            if (res == null)
+                throw new IllegalArgumentException("Couldn't fetch access value for key " + key);
+
+            return res;
+        }
+
+        private AcceptWay put(String key, Access access) {
+            accessMap.put(key, access);
+            if (access != Access.CAN_SKIP)
+                hasAccepted = true;
+            return this;
+        }
+
+        public boolean isEmpty() {
+            return accessMap.isEmpty();
+        }
+
+        public boolean hasAccepted() {
+            return hasAccepted;
+        }
+
+        private boolean has(String key) {
+            return accessMap.containsKey(key);
+        }
+    }
+
+    public enum Access {
+        WAY, FERRY, OTHER, CAN_SKIP;
+
+        boolean isFerry() {
+            return this.ordinal() == FERRY.ordinal();
+        }
+
+        boolean isWay() {
+            return this.ordinal() == WAY.ordinal();
+        }
+
+        boolean isOther() {
+            return this.ordinal() == OTHER.ordinal();
+        }
+
+        boolean canSkip() {
+            return this.ordinal() == CAN_SKIP.ordinal();
```

Comment:
Why canSkip?

Reply:
like a way for houses

Reply:
Thanks for explaining. Maybe this is just me, but I always thought the `return 0 ` to be more a `don't accept ` than a `can skip`, that's why I asked.

Reply:
"can skip" in the sense of do not use it for this flagencoder. So it could be used for another ... still we can use a different wording like "ignore"?

Reply:
Thanks, yeah maybe can skip is not bad from this point of view. Maybe ignore would be better, but I am not sure, I think we can also leave it as it is right now.